apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -565,6 +596,30 @@ export default function manageStudents(state = initialState, action) {
     }
     return newState;
   }
+  if (action.type === ADD_STUDENT_FULL) {
+    let newState = {
+      ...state,
+      addStatus: {
+        ...state.addStatus,
+        status: AddStatus.FULL,
+        numStudents: action.numStudents,
+        sectionCapacity: action.sectionCapacity,
+        sectionCode: action.sectionCode,
+        sectionStudentCount: action.sectionStudentCount
+      }
+    };
+    for (let i = 0; i < action.studentIds.length; i++) {
+      newState.studentData[action.studentIds[i]] = {
+        ...state.studentData[action.studentIds[i]],
+        isSaving: false
+      };
+      newState.editingData[action.studentIds[i]] = {
+        ...state.editingData[action.studentIds[i]],
+        isSaving: false
+      };
+    }
+    return newState;
+  }
```

Comment:
For my own understanding, what's this loop doing?

Reply:
Sorry for the delay!

This loop I snagged from the `ADD_STUDENT_FAILURE` case. It sets the status of each of the students that failed to add with `{ isSaving: false}`, which allows a user to continue to make edits without refreshing the page. 

Without this loop, an attempt at `save all` disables the individual `save` buttons for a failed bulk add. A user would need to refresh the page and resubmit a bulk add to try again.

With it, the full notification displays, and teachers can add individual students with the `add` buttons, or cancel students to match the spaces available and re-submit a bulk add without refreshing the page.

Hopefully, that makes sense? happy to demo it as well if that's helpful!